controller/appcontroller.go

```diff
@@ -836,6 +837,13 @@ func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Applic
 	config := metrics.AddMetricsTransportWrapper(ctrl.metricsServer, app, cluster.RESTConfig())
 
 	filteredObjs := FilterObjectsForDeletion(objs)
+
+	if app.GetPropagationPolicy() == appv1.BackgroundPropagationPolicy {
+		return ctrl.backgroundDeletion(app, objs, filteredObjs, logCtx, config)
+	}
+
+	logCtx.Info("Deleting application with foreground propagation policy")
```

Comment:
Personally, I like being able to know which thing is being poked, ... could you include <for {app name}>?